xbmc/network/GUIDialogNetworkSetup.cpp

```diff
@@ -191,20 +207,23 @@ void CGUIDialogNetworkSetup::InitializeSettings()
   // Add our protocols
   TranslatableIntegerSettingOptions labels;
 #ifdef HAS_FILESYSTEM_SMB
-  labels.push_back(std::make_pair(20171, NET_PROTOCOL_SMB));
+  m_protocols.emplace_back(Protocol{true, true, true, false, true, 0, "smb", 20171});
 #endif
-  labels.push_back(std::make_pair(20301, NET_PROTOCOL_HTTPS));
-  labels.push_back(std::make_pair(20300, NET_PROTOCOL_HTTP));
-  labels.push_back(std::make_pair(20254, NET_PROTOCOL_DAVS));
-  labels.push_back(std::make_pair(20253, NET_PROTOCOL_DAV));
-  labels.push_back(std::make_pair(20173, NET_PROTOCOL_FTP));
-  labels.push_back(std::make_pair(20175, NET_PROTOCOL_UPNP));
-  labels.push_back(std::make_pair(20304, NET_PROTOCOL_RSS));
+  const std::vector<Protocol> defaults =
+        {{ true,  true,  true,  true, false, 443, "https", 20301},
+         { true,  true,  true,  true, false,  80,  "http", 20300},
+         { true,  true,  true,  true, false, 443,  "davs", 20254},
+         { true,  true,  true,  true, false,  80,   "dav", 20253},
+         { true,  true,  true,  true, false,  21,   "ftp", 20173},
+         {false, false, false, false,  true,   0,  "upnp", 20175},
+         { true,  true,  true,  true, false,  80,   "rss", 20304}};
+
+ m_protocols.insert(m_protocols.end(), defaults.begin(), defaults.end());
 #ifdef HAS_FILESYSTEM_NFS
-  labels.push_back(std::make_pair(20259, NET_PROTOCOL_NFS));
+  m_protocols.emplace_back(Protocol{true, false, false, false, true, 0, "nfs", 20259});
 #endif
 #ifdef HAS_FILESYSTEM_SFTP
-  labels.push_back(std::make_pair(20260, NET_PROTOCOL_SFTP));
+  m_protocols.emplace_back(Protocol{true, true, true, true, false, 22, "sftp", 20260});
 #endif
 
   AddSpinner(group, SETTING_PROTOCOL, 1008, SettingLevel::Basic, m_protocol, labels);
```

Comment:
@notspiff `labels` now doesn't contain any elements and therefore the Protocol spinner doesn't show up